modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeTopologyService.java

```diff
@@ -59,6 +59,9 @@ final class ScaleCubeTopologyService extends AbstractTopologyService {
     /** Topology members map from the consistent id to the cluster node. */
     private final ConcurrentMap<String, ClusterNode> consistentIdToMemberMap = new ConcurrentHashMap<>();
 
+    /** Topology members map from the non-consistent id to the cluster node. */
+    private final ConcurrentMap<String, ClusterNode> idToMemberMap = new ConcurrentHashMap<>();
```

Comment:
Why don't you scan consistentIdToMemberMap instead of creating the new collection?

Reply:
There is a TODO mentioning https://issues.apache.org/jira/browse/IGNITE-20408, it will have to be removed later anyway.

Reply:
Anyway, I removed this to reduce the number of changes.